### DISCUSSION TITLE
variogramvario_estimate(direction..)

### DISCUSSION THREAD

**Discussion by cecidip** [GeoStat-Framework/GSTools#236] [Q&A] [answered]
Hello, i am trying to find an optimal angle for my anisotropy. The description sais " _If multiple directions are given, a set of variograms will be returned_" , so far i could only make it work as in the [example](https://geostat-framework.readthedocs.io/projects/gstools/en/stable/examples/03_variogram/03_directional_2d.html?highlight=variogram%202d#directional-variogram-estimation-and-fitting-in-2d): 
 direction=gs.rotated_main_axes(dim=2, angles=angle)

But if i put more than the main and orthogonal directions it does not work. 

How can I input several angles then ?

Sorry for the silly question and thank you,

Cecilia

**Comment by MuellerSeb** [comment]
Hey there!

There are no silly questions.
Could you provide a minimal working example and the exact outcome or error?

Cheers, Sebastian

**Comment by MuellerSeb** [comment, marked as answer]
For example, this is working for me:
```python
import numpy as np
from matplotlib import pyplot as plt

import gstools as gs

angle = np.pi / 8
model = gs.Exponential(dim=2, len_scale=[10, 5], angles=angle)
x = y = range(101)
srf = gs.SRF(model, seed=123456)
field = srf((x, y), mesh_type="structured")

# small search steps up to 90 degree
search_angles = [i * np.pi / 8 for i in range(5)]
directions = [[np.cos(ang), np.sin(ang)] for ang in search_angles]

bin_center, varios = gs.vario_estimate(
    (x, y),
    field,
    mesh_type="structured",
    direction=directions,
)

for var, ang in zip(varios, search_angles):
    plt.plot(bin_center, var, label=f"{np.rad2deg(ang)}")
plt.legend()
plt.show()
```
![Figure_1](https://user-images.githubusercontent.com/19690642/164242774-34a3bf2c-2f64-4f61-a2c0-8ed64114ac71.png)

**Reply by cecidip** [reply]
Sorry for the delay replying. Thank you , this worked  !
